### ISSUE TITLE
Checked + disabled fails 4.5 : 1 color contrast

### ISSUE THREAD

**Issue by jasonkaedingICE** [18F/web-design-standards#1830]
## Description

First, I was glad to see the checked + disabled state styling was implemented in USWDS.  However, we have received a 508 failure on the color contrast between the grey box and the white check.  Current colors result in a 1.4 : 1 contrast ratio.

## Steps to reproduce the issue

```
<input id="option1" type="checkbox" checked="checked" disabled="disabled" value="something" />
<label for="option1">Something</label>
```

## Suggested change
Font for disabled radio/checkbox label is 508 compliant on white background (#5b616b). I used that color as a basis for the following change. Color contrast is now 4.76 : 1, and the border is a couple shades darker to match how the border was darker than the fill color in the current implementation.

```
[type=checkbox]:disabled + label::before,
[type=radio]:disabled + label::before {
  background: #6d7380;
  box-shadow: 0 0 0 1px #4f545d;
  cursor: not-allowed;
}
```

**Comment by shawnbot** [comment]
Thanks for reporting this. We've determined that disabled/inactive form elements don't have the same contrast requirements as interactive ones. See #757 for more info, and [this thread on UX Stack Exchange](http://ux.stackexchange.com/questions/80881/do-disabled-buttons-still-need-to-be-contrast-compliant-for-accessibility).

**Comment by jasonkaedingICE** [comment]
@shawnbot I can certainly see the rationale. In our case perhaps it is a bit more nuanced and I figured I'd share in case it is relevant to future design considerations. 

In our application we have a group of checkboxes representing a "complete" package. Some of these items are manually performed and some are automatically performed via system interfaces. We used checked and disabled to communicate the system-derived items. So it actually does communicate useful information in this situation (whether the item is present) but we didn't want to make it seem "clickable" in order to not confuse the user.

**Comment by shawnbot** [comment]
That's an interesting challenge. Maybe a non-interactive element (an image, say) with [`aria-checked="true"`](https://www.w3.org/TR/wai-aria/states_and_properties#aria-checked) would allow you to show an image and provide more informative cues to non-sighted users?

Paging our resident accessibility expert @nickbristow 🚨

**Comment by nickbristow** [comment]
@jasonkaedingICE @shawnbot I actually was just discussing this with someone else. The issue with using disabled form fields to convey information is they are completely ignored by screen readers. 

I would definitely look at alternate solutions for displaying that information. Shawn's suggestion would work.

**Comment by jasonkaedingICE** [comment]
@nickbristow thanks for the info. We actually have a couple instances of a disabled input communicating info. In this case, we actually enable the checkboxes upon submit because they are part of the submission.  We have another case where we have disabled text fields paired with type="hidden" inputs to pass along the values. I guess in both patterns the values are part of the submission but immutable.

This got me thinking, would a <label> with a for attribute pointing to a hidden input be a way to approach that? I know we had a couple of completely orphaned labels that failed 508 but I wasn't sure how the screen reader would treat this pattern since it's at least linked to something.

**Comment by nickbristow** [comment]
I don't think i've ever tested that pattern. I would assume it would get pretty inconsistent results.

**Comment by jasonkaedingICE** [comment]
@nickbristow our 508 tester just went through with the screen reader as-is (disabled+checked) and she reports that the NVDA screen reader reads "[label] Unavailable".  Obviously this is just one screen reader and others may behave differently.  I just figured I'd share. Even so, it isn't reading whether it's checked or not which is part of the meaningful information.
